R/pretty_colollipop_plot.R

```diff
@@ -0,0 +1,195 @@
+    dim1 <- dim(ssm1)[1]  
+    dim2 <- dim(ssm2)[1]
+
+    if(dim1 == 0 | dim2 == 0) {
+        stop(paste0("Ensure all variants in metadata are accounted for. \nEither ensure all variants are loaded into metadata as one dataframe, or subset your metadata so that all variants are included."))
+    } 
+
+    # Arguments to pass into pretty_lollipop_plot
+    lollipopplot_args <- list(...)
+
+    # Get gene_counts data for the first plot (without generating the plot)
+    lp1_gene_counts <- do.call(pretty_lollipop_plot, c(
+        list(
+            maf_df = ssm1,
+            gene = gene,
+            plotarg = FALSE
+        ), 
+        lollipopplot_args
+    ))
+    lp1_gene_counts_data <- as.data.frame(lp1_gene_counts)
+
+    # Get gene_counts data for the second plot
+    lp2_gene_counts <- do.call(pretty_lollipop_plot, c(
+        list(
+            maf_df = ssm2,
+            gene = gene,
+            plotarg = FALSE
+        ), 
+        lollipopplot_args
+    ))
+    lp2_gene_counts_data <- as.data.frame(lp2_gene_counts)
+
+    # Combine data for both plots
+    lp1_gene_counts_data <- lp1_gene_counts_data %>% 
+        mutate(source = "Sample 1")
+    lp2_gene_counts_data <- lp2_gene_counts_data %>% 
+        mutate(source = "Sample 2")
+
+    combined_gene_counts <- rbind(
+        lp1_gene_counts_data, 
+        lp2_gene_counts_data 
+    )
+
+    # Setting Somatic Mutation Statistic
+    meta1_counter <- length(unique(meta1$Tumor_Sample_Barcode))
+    meta2_counter <- length(unique(meta2$Tumor_Sample_Barcode))
+
+    # Pass combined_gene_counts to pretty_lollipop_plot for plotting
+    colollipop_plot <- pretty_lollipop_plot(
+        maf_df = maf_df, 
+        gene = gene, 
+        refseq_id,
+        plot_title = plot_title,
+        include_silent = FALSE,
+        label_threshold = label_threshold,
+        plotarg = TRUE,
+        mirrorarg = TRUE,
+        combined_gene_counts = combined_gene_counts,
+        meta1_counter = meta1_counter,
+        meta2_counter = meta2_counter,
+        Sample1 = comparison_values[1],
+        Sample2 = comparison_values[2]
+    )
+
+    forest_plot <- prettyForestPlot(
+        maf = maf_df,
+        metadata = metadata,
+        genes = gene,
+        comparison_column = comparison_column,
+        comparison_values = comparison_values,
+        separate_hotspots = FALSE,
+        comparison_name = paste0(
+            comparison_values[1],
+            " vs. ",
+            comparison_values[2]
+        )
+    )
+
+    # arrange colollipop plot and forest plot together one over the other
+    forest_plot <- ggarrange(
+        colollipop_plot,              
+        ggarrange(
+            forest_plot$forest,        
+            forest_plot$bar, 
+            widths = c(1, 0.6),
+            common.legend = TRUE,
+            align = "h"),          
+        ncol = 1, 
+        nrow = 2,
+        heights = c(3, 1)            
+    )
+
+    if (forestarg == TRUE) {
```

Comment:
There are some issues with the final layout being quite compressed with overlapping text, even when printing to a plot that is 10" wide by 15" high. Also prettyForestPlot uses theme_Morons (arial font) while the rest of the plot uses the default Helvetica - perhaps best to change the theme for the forest_plot to match so the font is consistent.